packages/tpetra/core/src/Tpetra_DirectoryImpl_def.hpp

```diff
@@ -452,7 +470,7 @@ namespace Tpetra {
           const GO one = as<GO> (1);
           const GO two = as<GO> (2);
           const GO nOverP_GID = as<GO> (nOverP);
-          const GO lowerBound = GID / std::max(nOverP_GID, one) + two;
+          const GO lowerBound = i0 + GID / std::max(nOverP_GID, one) + two;
```

Comment:
Why did Tpetra add `two` here?  (I actually don't know; I just found this here and fixed the casting mistakes long ago.)  It seems like this change could make the `while` loop below break early, claiming that it couldn't find the GID.  We could just delete the `two`; the loop should still work fine.

Reply:
i don't know why the two is there either.  it is also in the Epetra version.

Reply:
when i changed the search to use a bisection it broke a number of other Tpetra tests.  so, I just left the search as is, only changing the lower bound of where the search could start.

Reply:
So, I removed the `two`, but a Albany labeled Tpetra test broke.  I didn't look in to it further.